qiskit/providers/backend.py

```diff
@@ -229,3 +233,239 @@ class can handle either situation.
             Job: The job object for the run
         """
         pass
+
+
+class BackendV2(Backend, ABC):
+    """Abstract class for Backends
+
+    This abstract class is to be used for all Backend objects created by a
+    provider. This version differs from earlier abstract Backend classes in
+    that the configuration attribute no longer exists instead attributes
+
+    The ``options`` attribute of the backend is used to contain the dynamic
+    user configurable options of the backend. It should be used more for
+    runtime options that configure how the backend is used. For example,
+    something like a ``shots`` field for a backend that runs experiments which
+    would contain an int for how many shots to execute.
+
+    If migrating a provider from :class:`~qiskit.providers.BackendV1` or
+    :class:`~qiskit.providers.BaseBackend` one thing to keep in mind is for
+    backwards compatibility you might need to add a configuration method that
+    will build a :class:`~qiskit.providers.models.BackendConfiguration` object
+    and :class:`~qiskit.providers.models.BackendProperties` from the attributes
+    defined in this class for backwards compatibility.
+    """
+
+    version = 2
+
+    def __init__(self, provider, **fields):
+        self._options = self._default_options()
+        self._provider = provider
+        self._coupling_map = None
+        if fields:
+            for field in fields:
+                if field not in self._options.data:
+                    raise AttributeError("Options field %s is not valid for this backend" % field)
+            self._options.update_config(**fields)
+
+    @property
+    def instructions(self) -> List[Gate]:
+        """A list of Gate instances that the backend supports."""
+        return list(self.target.gates)
+
+    @property
+    def instructions_names(self) -> List[str]:
+        """A list of Gate names that the backend supports."""
+        return list(self.target.gate_names)
+
+    @property
+    @abstractmethod
+    def target(self):
+        """A :class:`qiskit.transpiler.Target` object for the backend."""
+        pass
+
+    @property
+    @abstractmethod
+    def num_qubits(self) -> int:
+        """Return the number of qubits the backend has."""
+        pass
+
+    @property
+    def coupling_map(self):
+        """Return the :class:`~qiskit.transpiler.CouplingMap` object"""
+        return self.target.coupling_map()
+
+    @property
+    def instruction_durations(self):
```

Comment:
This is not quite universal because, for example, noise-less circuit simulator doesn't need to define gate duration.

Reply:
That's true I just left it in here because `Target` can generate an `InstructionDurations` object. But if your target is a noisless simulation I would expect this to just return an empty `InstructionDurations` object. But I think this raises a good point, I need to think some more about what a Target object would look like in a noiseless simulator and maybe add some handling for that case (like allowing `None` for the qargs dict so aer doesn't need to specify all qubits for every gate it supports or something).

Reply:
I left these as properties of the `BackendV2` class, but methods on the Target class (although looking now I think that I missed the instruction schedule map for the backend class). My thinking was for the target we're generating a view object (although not strictly because it's actually a copy) with basically a backwards compatibility type for information the target contains, but for a backend object they're attributes of the backend (like the target itself is), but because the target contains the information there is a default implementation that just queries the target to the get the CouplingMap object for example.

That being said I can change this here to be methods too, it doesn't bother me either way I just thought attribute access was a bit cleaner for the `BackendV2` class.

Reply:
I just feel for a user this distinction is not obvious to quickly know when to put `()` or not. But I don't feel too strongly about either way.